opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1451,25 +1488,35 @@ namespace Opm {
 
 
     template<typename TypeTag>
-    std::tuple<bool, bool, double>
+    std::pair<bool, bool>
     BlackoilWellModel<TypeTag>::
-    updateWellControls(DeferredLogger& deferred_logger)
+    updateWellControls(DeferredLogger& deferred_logger,
+                       const std::size_t network_update_it)
     {
         // Even if there are no wells active locally, we cannot
         // return as the DeferredLogger uses global communication.
         // For no well active globally we simply return.
-        if( !wellsActive() ) return { false, false, 0.0 };
+        if( !wellsActive() ) return { false, false };
 
         const int episodeIdx = ebosSimulator_.episodeIndex();
         const int iterationIdx = ebosSimulator_.model().newtonMethod().numIterations();
         const auto& comm = ebosSimulator_.vanguard().grid().comm();
         updateAndCommunicateGroupData(episodeIdx, iterationIdx);
 
-        const auto [local_network_changed, local_network_imbalance]
-            = shouldBalanceNetwork(episodeIdx, iterationIdx) ?
-            updateNetworkPressures(episodeIdx) : std::make_pair(false, 0.0);
-        const bool network_changed = comm.sum(local_network_changed);
-        const double network_imbalance = comm.max(local_network_imbalance);
+        // network related
+        bool more_network_update = false;
+        if (shouldBalanceNetwork(episodeIdx, iterationIdx)) {
+            const auto [local_network_changed, local_network_imbalance] = updateNetworkPressures(episodeIdx);
+            const bool network_changed = comm.sum(local_network_changed);
+            const double network_imbalance = comm.max(local_network_imbalance);
+            if (network_changed) {
+                const auto& balance = schedule()[episodeIdx].network_balance();
+                // Iterate if not converged, and number of iterations is not yet max (NETBALAN item 3).
+                if (network_update_it < balance.pressure_max_iter() && network_imbalance > balance.pressure_tolerance()) {
```

Comment:
I changed the code to keep the function with a slightly different name `moreNetworkIteration`. 

It was not intentional to rewrite the code of the function instead of calling the function here.